src/Plugin.php

```diff
@@ -537,11 +537,36 @@ public function checkStates($scan_inactive_and_new_plugins = false)
         // Check all directories from the checklist
         foreach ($directories as $directory) {
             $this->checkPluginState($directory, $scan_inactive_and_new_plugins);
+            $this->updatePluginInformation($directory);
```

Comment:
This will trigger many SQL request on every script of GLPI. Updating plugins informations should probably be done only in the display process of the plugin admin/marketplace pages. There is already a specifi call in these pages to scan all the plugins directories: `$plugin->checkStates(true);`.